server/server.js

```diff
@@ -113,48 +143,87 @@ async function getUserIdFromSub(userSub) {
     return user.rows[0].user_id;
 }
 
+
 //update and add feed events
 app.post("/api/feed", async (req, res) =>  {
     
     try {
-        console.log(req.body)
-        const {auth0_sub, api_id, isFav } = req.body;
+        
+        const {auth0_sub, api_id, isFav, shelf_status } = req.body;
+
         let user_id = await getUserIdFromSub(auth0_sub)
 
         const existingEntry = await db.query("SELECT * FROM feeds WHERE api_id = $1 AND user_id = $2", [api_id, user_id]);
 
         if (existingEntry.rows.length > 0) {
             const updatedBook = await db.query(
                 "UPDATE feeds SET isFavorite = $1, shelf_status = $2, note = $3 WHERE api_id = $4 AND user_id = $5 RETURNING *",
-                [isFav, null, null, api_id, user_id]
+                [isFav, shelf_status, null, api_id, user_id]
             );
+            // Send back the updated book entry with a 200 OK status
+            res.status(200).json(updatedBook.rows[0]);
         }
         else {
             const newItem = await db.query(
-                "INSERT INTO feeds (api_id, user_id, isfavorite, shelf_status, note) VALUES ($1, $2, $3, $4, $5) RETURNING *", [api_id, user_id, isFav, null, null]
+                "INSERT INTO feeds (api_id, user_id, isfavorite, shelf_status, note) VALUES ($1, $2, $3, $4, $5) RETURNING *", [api_id, user_id, isFav, shelf_status, null]
             );
+
+            // Send back the new feed entry with a 201 Created status
+            res.status(201).json(newItem.rows[0]);
         }
-        res.json({});
+
     } catch (error) {
         console.error("Error Message!:", error.message);
+        res.status(500).json({ message: error.message });
     }
 
 });
 
 
-//queries all the user actions
+//queries for the user's all actions
 app.get("/api/feed/:id", async (req, res) =>  {
     
     try {
-        // const {auth0_sub} = req.body;
         const { id } = req.params;
         console.log(id)
 
         let user_id = await getUserIdFromSub(id)
         const {rows : user_actions} = await db.query('SELECT * FROM feeds WHERE user_id = $1', [user_id]);
-        res.send(user_actions);
+
+        if (user_actions.length === 0) {
+            res.status(200).json([]); 
+        } 
+        else {
+            res.status(200).json(user_actions);
+        }
```

Comment:
wouldn't `user_actions` be an empty array anyways? I don't think you need the `if` statement